<launch>
  <!-- stereo preprocessing -->
  <arg name="INPUT_POINT_CLOUD" default="/multisense/resize_1_4/points" />
  <!-- compute normal -->
  <node pkg="nodelet" type="nodelet" name="preprocessing_manager"
        args="manager" output="screen"/>
  <node pkg="nodelet" type="nodelet" args="load jsk_topic_tools/Relay preprocessing_manager"
        name="relay">
    <remap from="~input" to="$(arg INPUT_POINT_CLOUD)" />
  </node>
  <node pkg="nodelet" type="nodelet" args="load jsk_pcl/NormalEstimationIntegralImage  preprocessing_manager"
        name="normal_estimation">
    <remap from="~input" to="relay/output" />
  </node>
  <node pkg="nodelet" type="nodelet" args="load jsk_pcl/TfTransformCloud preprocessing_manager"
        name="stereo_ground_cloud">
    <remap from="~input" to="relay/output" />
    <rosparam>
      target_frame_id: ground
    </rosparam>
  </node>
  <node pkg="nodelet" type="nodelet" args="load jsk_topic_tools/Relay preprocessing_manager"
        name="stereo_downsampled_ground_cloud">
    <remap from="~input" to="stereo_ground_cloud/output" />
  </node>
  <node pkg="nodelet" type="nodelet" args="load pcl/VoxelGrid preprocessing_manager"
        if="false"
        name="stereo_downsampled_ground_cloud">
    <remap from="~input" to="stereo_ground_cloud/output" />
    <rosparam>
      leaf_size: 0.02
      filter_limit_min: -100
      filter_limit_max: 100
    </rosparam>
  </node>
</launch>
